onnxruntime/core/optimizer/selectors_actions/selector_action_transformer.h

```diff
@@ -109,7 +113,15 @@ class SelectorActionTransformer : public GraphTransformer {
   SelectorsAndActions selectors_and_actions_;
 
 #if !defined(ORT_MINIMAL_BUILD)
-  Status MatchAndProcess(Graph& graph, Node& node, bool& modified, const logging::Logger& logger) const;
+
+  // check if the node matches any of the registered operators.
+  // if it does, run the Selector.
+  // if that selects nodes, run the Action.
+  //
+  // Note, MatchAndProcess takes both Graph and GraphViewer to avoid expensive recreate of GraphViewer,
+  // the graph must be the same as the graph_viewer's underlying graph
+  Status MatchAndProcess(Graph& graph, const GraphViewer& graph_viewer, Node& node,
```

Comment:
Would be good to call out why we need to use a GraphViewer for selection and not the Graph instance. otherwise the reason for the setup having both a Graph and GraphViewer is unclear and people will think you were insane for making this change :P